OneSignalSDK/onesignal/core/src/main/java/com/onesignal/user/internal/operations/impl/executors/IdentityOperationExecutor.kt

```diff
@@ -67,11 +69,15 @@ internal class IdentityOperationExecutor(
                     NetworkUtils.ResponseStatusType.UNAUTHORIZED ->
                         ExecutionResponse(ExecutionResult.FAIL_UNAUTHORIZED)
                     NetworkUtils.ResponseStatusType.MISSING -> {
-                        val operations = _buildUserService.getRebuildOperationsIfCurrentUser(lastOperation.appId, lastOperation.onesignalId)
-                        if (operations == null) {
+                        if (_newRecordState.isInMissingRetryWindow(lastOperation.onesignalId)) {
+                            return ExecutionResponse(ExecutionResult.FAIL_RETRY)
```

Comment:
It looks like all responses with status type of MISSING (404, 410) will eventually be handled as either FAIL_NORETRY, FAIL_RETRY, or SUCCESS. If we get a 410, shouldn't it always be no retry?

Reply:
hmm, that is a good point. 404 simply means that the URI you are accessing doesn't exist, where generally a 410 means the server knew it did exist at some point, and now it is gone.
* I don't think the backend does this now, but it's always possible that the backend could keep track of the delete to give us this request.

Reply:
@jinliu9508 I addressed this in a new commit, by checking if it was a 404 here.

Reply:
This would work. Just a nit, since 404 and 410 will be handled differently now, I would suggest adding a new ResponseStatusType in NetworkUtils just for 410 so we can handle it separately and don't need to specifically watch for it with 404 handling, which makes the code more intuitive and slightly efficient. This is really small and doesn't have to be in this PR. I will resolve it and feel free to make any more change regarding this nit.